### ISSUE TITLE
[Suggestion] Hotbar should prioritize small stacks

### ISSUE THREAD

**Issue by evie-calico** [blackd/Inventory-Profiles#304]
At the moment, pick-blocking or shift-clicking will first try to combine a stack into the player's hotbar, before moving into the proper inventory.

As an example, shift-clicking this cobblestone:
![image](https://user-images.githubusercontent.com/14899090/225773662-0f6e1c3e-e571-4052-89a7-4ea44278a25c.png)

becomes this:
![image](https://user-images.githubusercontent.com/14899090/225773675-eff2dbfd-dfbe-435b-b706-971c4e9e4dd9.png)

However, it would be more convenient if the smallest stack was placed in the hotbar, like this:
![image](https://user-images.githubusercontent.com/14899090/225773708-2fdac1bc-c29c-426c-9da8-4771e67b5a3f.png)

This way smaller stacks can be used up first and free up an inventory slot, as opposed to ending up with two incomplete stacks in both the hotbar and inventory.

Similarly, it would be nice if pick-blocking and the "auto refill" that occurs when you run out of blocks grabbed the smallest stack.

**Comment by blackd** [comment]
Hi,

Shift click is handled by minecraft in particular it's handled server side. If I reimplement it to be handled client side this will break lots of server plugins that introduce special behaviour when shift clicking. I don't really wonna do that because it will lead to too many complains. Also this will break lots of mods that add new equipment slots for example trinkets. 

About keeping full stacks in the inventory and refilling with partial. This will require complete rewrite of the code that handles auto refill. I'll look into this and if it's not too complex I'll add an option in the settings. 

TLDR; 
Change shift click - No. 
Refilling with partial stacks first - May be.

**Comment by evie-calico** [comment]
I figured this might be more complicated than it seems. This is a pretty minor request anyways so it's completely understandable if it can't be reasonably implemented :)
